src/datajudge/formatter.py

```diff
@@ -0,0 +1,50 @@
+import abc
+import re
+
+from colorama import Back
+
+STYLING_CODES = r"\[(numMatch|numDiff)\](.*?)\[/\1\]"
+
+
+class Formatter(abc.ABC):
+    @abc.abstractmethod
+    def fmt_str(self, result: str) -> str:
+        pass
+
+
+class DefaultFormatter(Formatter):
```

Comment:
Mostly stylistic, but I'd put the implementation of the `__init__` and `fmt_str` directly in the base class and then let all specific implementations derive from that one instead of the `DefaultFormatter`.